flutter/lib/src/localizations.dart

```diff
@@ -0,0 +1,11 @@
+import 'package:flutter/cupertino.dart';
+import 'package:meta/meta.dart';
+
+@internal
+Locale? retrieveWidgetLocale(GlobalKey<NavigatorState>? navigatorKey) {
+  final BuildContext? context = navigatorKey?.currentContext;
+  if (context != null) {
+    return Localizations.maybeLocaleOf(context);
+  }
+  return null;
+}
```

Comment:
I guess this could be added directly in the `FlutterEnricherEventProcessor`, since this is unlikely to be used somewhere else.